### PR TITLE
print more information on command failure

### PR DESCRIPTION
**Pull request by charlierudolph**
Examples:

```
$ git town set-parent-branch
Feature branches can be branched directly off
master or from other feature branches.

The former allows to develop and ship features completely independent of each other.
The latter allows to build on top of currently unshipped features.


Error: 
----------------------------------------
Diagnostic information of failed command

Command: git branch2
Error: exit status 1
Output:
git: 'branch2' is not a git command. See 'git --help'.

The most similar command is
        branch

----------------------------------------


```

And

```
$ git hack test
[cr-diagnostic] git fetch --prune --tags

[cr-diagnostic] git add -A

[cr-diagnostic] git stash
Saved working directory and index state WIP on cr-diagnostic: 614b51eb update

[cr-diagnostic] git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.

[master] git rebase origin/master
Current branch master is up to date.

[master] git branch test master

[master] git checkout test
Switched to branch 'test'

Error: cannot determine whether the local branch "master" exists: 
----------------------------------------
Diagnostic information of failed command

Command: git branch2
Error: exit status 1
Output:
git: 'branch2' is not a git command. See 'git --help'.

The most similar command is
        branch

----------------------------------------

To abort, run "git-town abort".
To continue after having resolved conflicts, run "git-town continue".

```

### CODE REVIEW

File: src/run/run_test.go [hunk 1]
```diff
@@ -26,10 +26,19 @@ func TestRun_Exec_UnknownExecutable(t *testing.T) {
...
-	var execError *exec.ExitError
-	assert.True(t, errors.As(err, &execError))
-	assert.Equal(t, 2, execError.ExitCode())
+	result, err := run.Exec("bash", "-c", "echo 'hi' && exit 2")
+	assert.Equal(t, 2, result.ExitCode())
+	expectedError := `
+----------------------------------------
+Diagnostic information of failed command
+
+Command: bash -c echo 'hi' && exit 2
```
Comment (kevgo) [inline src/run/run_test.go:35]: This should be okay without the single quotes since `hi` contains no quotable characters.

**Review by kevgo — APPROVED** [review]
I love the addition of exitCode to Result! Thanks for doing this! In a follow-up PR, we should add more diagnostic information to the output that allows to reproduce issues, like Git/Git Town/System version/configuration etc.

File: src/run/result.go [hunk 2]
```diff
@@ -25,6 +26,11 @@ func (c *Result) Command() string {
 	return c.command
 }
 
+// ExitCode returns the exit code of the command
```
Comment (kevgo) [inline src/run/result.go:29]: Please add a period to the end of this comment.

File: src/run/run_test.go (hunk 1 above)
Comment (charlierudolph) [reply src/run/run_test.go]: Removed quotes

**Pull request merged**